MyLibrary/Library/Controllers/BookController.cs

```diff
@@ -0,0 +1,157 @@
+        }
+
+        // GET: api/book
+        /// <summary>
+        /// Gets all books from the Library
+        /// </summary>
+        /// <returns>
+        /// HTTP status code Ok() with the list of books
+        /// or HTTP status code NoContent() if there are any books in library
+        /// </returns>
+        [HttpGet]
+        public IActionResult GetAllBooks()
+        {
+            if (this.libraryService.GetAllBooks() != null)
+            {
+                return Ok(this.libraryService.GetAllBooks());
+            }
+
+            return NoContent();
+        }
+
+        // GET: api/book/5
+        /// <summary>
+        /// Gets the book by its id
+        /// </summary>
+        /// <param name="id">The identifier of the book</param>
+        /// <returns>
+        /// HTTP status code Ok() with found book
+        /// or HTTP status code NotFound() if there isn`t book with such id
+        /// </returns>
+        [HttpGet("{id}")]
+        public IActionResult GetBook(uint id)
+        {
+            Book foundBook = this.libraryService.GetBookById(id);
+            if (foundBook == null)
+            {
+                return NotFound("No books with such id!");
+            }
+
+            return Ok(foundBook);
+        }
+
+        // POST: api/book
+        /// <summary>
+        /// Adds the new book to the book list
+        /// </summary>
+        /// <param name="book">The book for adding</param>
+        /// <returns>
+        /// HTTP status code Created() with added book`s id
+        /// or HTTP status code BadRequest() with error message
+        /// </returns>
+        [HttpPost]
+        public IActionResult AddNewBook([FromBody]Book book)
+        {
+            // ???
+            if (!ModelState.IsValid)
+            {
+                return BadRequest("Invalid book input");
+            }
+
+            try
+            {
+                uint newBookId = this.libraryService.AddBook(book);
+                return Created("book", newBookId);
+            }
+            catch (ArgumentException exception)
+            {
+                return BadRequest(exception.Message);
+            }
+        }
+
+        // PUT: api/book/5
+        /// <summary>
+        /// Updates the book
+        /// </summary>
+        /// <param name="bookId">The book identifier</param>
+        /// <param name="newBookName">New name of the book</param>
+        /// <param name="newBookYear">The new publising year of the book</param>
+        /// <param name="newAuthorId">The new author identifier of the book</param>
+        /// <returns>
+        /// HTTP status code Ok() with updated book`s id
+        /// or HTTP status code NotFound() if there is no books with specified id
+        /// or HTTP status code BadRequest if there is no authors with specified new id</returns>
+        [HttpPut("{id}")]
+        public IActionResult UpdateBook(uint bookId, string newBookName, int newBookYear,
```

Comment:
Why u don't [FromBody] book for inputed params like it was with authors?

Reply:
I wanted to try different variants